packages/eas-cli/src/metadata/utils/retry.ts

```diff
@@ -0,0 +1,28 @@
+export async function waitAsync(duration: number): Promise<void> {
+  return new Promise(resolve => setTimeout(resolve, duration));
+}
+
+type WithRetryOptions = {
+  tries?: number;
+  delay?: number;
+  onRetry?: (triesLeft: number) => void;
+};
+
+export async function retryIfNullAsync<T>(
```

Comment:
I would write a generic retry function and move it to common utils for cli
retrying when the result is null is a bit weird, so probably a cleaner version to handle that would be on call site
e.g.
```
await retryAsync(async () => {
   const result = app.getEditAppStoreVersionAsync({ platform });
   if (!resul) {
     throw new Error('some not found message')
   }
}, retryOptions)
```
also you probably want to retry if `method` throws an exception